src/tape/ensemble_frame.py

```diff
@@ -924,6 +947,57 @@ def update_ensemble(self):
         # Update the Ensemble to track this frame and return the ensemble.
         return self.ensemble.update_frame(self)
 
+    @classmethod
+    def from_dict(
+        cl, data, npartitions, orient="columns", dtype=None, columns=None, label=None, ensemble=None
+    ):
+        """"""
+        result = from_dict(
+            data, npartitions=npartitions, orient=orient, dtype=dtype, columns=columns, constructor=TapeFrame
```

Comment:
Do we need an implementation of `from_dict` for `SourceFrame` and `ObjectFrame` as well? It looks like we only test `EnsembleFrame.from_dict` in test_ensemble_frame

May be able to resolve for all three methods with just `constructor=_partition_type`

Reply:
Thanks for this suggestion. Implemented and added some test coverage.